File: src/visual-graph/builder/node-id.test.ts
import { describe, expect, test } from "vitest";

import { nodeId } from "./node-id.js";

describe("nodeId", () => {
  test.each([
    { input: "foo123", expected: "n_foo123" },
    { input: "a.b-c", expected: "n_a_b_c" },
    { input: "", expected: "n_" },
  ])("nodeId($input) = $expected", ({ input, expected }) => {
    expect(nodeId(input)).toBe(expected);
  });
});

File: crates/unsnarl-visual-graph/src/builder/node_id_test.rs
//! Sibling tests for [`node_id`].

use super::node_id;

#[test]
fn alphanumerics_pass_through_with_prefix() {
    assert_eq!(node_id("foo123"), "n_foo123");
}

#[test]
fn specials_are_replaced_with_underscore() {
    assert_eq!(node_id("a.b-c"), "n_a_b_c");
}

#[test]
fn empty_input_yields_prefix_only() {
    assert_eq!(node_id(""), "n_");
}
